src/ppopt/utils/region_overlap_utils.py

```diff
@@ -0,0 +1,188 @@
+from itertools import permutations
+from typing import List, Tuple, Optional
+from collections import deque
+
+import numpy
+import copy
+
+from ..mpmilp_program import MPMILP_Program
+from ..critical_region import CriticalRegion
+
+from .mpqp_utils import get_bounds_1d
+
+
+def reduce_overlapping_critical_regions_1d(program: MPMILP_Program, regions: List[CriticalRegion]) -> Tuple[
+    List[CriticalRegion], bool]:
+    """
+    This adjusts critical regions for the 1d-parameter case, so that there are no overlaps.
+
+    :param program: the MPMILP_Program to be solved
+    :param regions: a list of CriticalRegion objects
+    :return: a list of CriticalRegion objects and a flag indicating whether there are still overlaps
+    """
+
+    # guard around the statement
+    if program.num_t() != 1:
+        raise ValueError('reduce_overlapping_critical_regions_1d requires a 1d-parameter problem')
+
+    overlaps_remaining, regions = identify_overlaps(program, regions)
+
+    return regions, overlaps_remaining
+
+
+def identify_overlaps(program: MPMILP_Program, regions: List[CriticalRegion]) -> Tuple[bool, list]:
+    new_regions = []
+    to_remove = []
+    possible_dual_degeneracy = False
+    # test all permutations rather than combinations: this way we only need to test for 2 cases (CR 2 fully inside CR 1, or CR 1 left of CR 2 but overlapping), since the other 2 cases are handled by the swapped permutation
+    to_check = deque(permutations(regions, 2))
+    while len(to_check) > 0:
+
+        region_added = False
+
+        cr1, cr2 = to_check.popleft()
+        # check if region has already been marked for removal, if so, no need to check again
+        if cr1 in to_remove or cr2 in to_remove:
+            continue
+
+        lb1, ub1 = get_bounds_1d(cr1.E, cr1.f)
```

Comment:
do we need to compute these bounds here or can we move them to the place in the code where we use them?